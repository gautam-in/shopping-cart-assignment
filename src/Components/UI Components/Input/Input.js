import "./Input.scss";

const Input = ({ input, handleChange, value }) => {
  const { inputId, placeholder, type } = input;
  return (
    <>
      <input
        type={type}
        className="input"
        id={inputId}
        placeholder={placeholder}
        name={inputId}
        value={value}
        onChange={handleChange}
        required
      />
      <label htmlFor={inputId} className="floating-label">
        {placeholder}
      </label>
    </>
  );
};

export default Input;
