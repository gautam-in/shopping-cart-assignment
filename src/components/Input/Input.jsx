import React from "react";
import PropTypes from "prop-types";
import styles from "./Input.module.scss";
import { useController } from "react-hook-form";

function Input(props) {

    const {
        label,
        name,
        type,
        ...rest
    } = props;

    const { field, fieldState } = useController(props);

    return (
        <div className={`${styles.input} ${fieldState.invalid ? styles.invalid : ""}`}>
            <div className={styles.inputWrapper}>
                <input type={type} name={name} placeholder="no-label" {...field} {...rest} />
                <label>{label}</label>
            </div>
            {
                fieldState.invalid ? <div className={styles.error} data-testid="input-error">{fieldState?.error?.message}</div> : null
            }
        </div>
    )
}

Input.propTypes = {
    label: PropTypes.string,
    value: PropTypes.any,
    onChange: PropTypes.func,
    errorMessage: PropTypes.string,
    name: PropTypes.string,
    type: PropTypes.oneOf(["text", "password"])
}

export { Input };
