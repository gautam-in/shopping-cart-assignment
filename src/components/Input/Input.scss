$sub-color: grey;
$main-color: black;

@mixin shrinkLabel {
  top: -14px;
  font-size: 12px;
  color: $main-color;
}

.input-container {
  position: relative;
  margin: 45px 0;

  .form-input {
    background: none;
    background-color: white;
    color: $sub-color;
    font-size: 16px;
    padding: 10px 0 10px 0;
    display: block;
    width: 100%;
    border: none;
    border-radius: 0;
    border-bottom: 1px solid $sub-color;
    margin: 25px 0;

    &:focus {
      outline: none;
      border-bottom: 1px solid #99dadf;
    }

    &:focus ~ .form-input-label {
      @include shrinkLabel();
    }
  }

  input[type="password"] {
    letter-spacing: 0.3em;
  }

  .form-input-label {
    color: $sub-color;
    font-size: 16px;
    font-weight: normal;
    position: absolute;
    pointer-events: none;
    left: 5px;
    top: 10px;
    transition: 300ms ease all;
    &.shrink {
      @include shrinkLabel();
      color: #99dadf;
    }
  }
}
