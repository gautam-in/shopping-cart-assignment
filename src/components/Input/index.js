import React from "react";
import "./style.scss";

function Input(props) {
  const { type, placeholder, maxLength, value, onChange, id, errorMessage } =
    props;
  return (
    <div className="input__wrapper">
      <input
        type={type}
        id={id}
        required
        value={value}
        onChange={onChange}
        max={maxLength}
      />
      <label htmlFor={id} className="input__label">
        {placeholder}{" "}
      </label>
      <p className="input--error">{errorMessage && errorMessage}</p>
    </div>
  );
}
export default Input;
// Input.defaultProps = {
//   type: "text",
//   placeholder: "Email",
//   maxLength: 60,
//   value="",
//   onChange="",
//   id: "",
//   errorMessage: "",
// };
