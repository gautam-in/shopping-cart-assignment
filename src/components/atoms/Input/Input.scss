.input-field {
  display: flex;
  flex-flow: column-reverse;
  margin-bottom: 1.5rem;
  position: relative;
}

label,
input {
  transition: all 0.2s;
  touch-action: manipulation;
}

input {
  font-size: 1.2rem;
  line-height: 1.6rem;
  font-weight: 500;
  border: 0;
  border-bottom: 0.2rem solid #eaeaea;
  border-radius: 0;
  padding: 0.8 0;
  cursor: text;
}

input:focus {
  outline: 0;
  color: #000000;
  border-bottom: 0.2rem solid #50dae2;
}

label {
  font-size: 1rem;
  line-height: 1.2rem;
  font-weight: 500;
  color: #999999;
  padding: 0.4rem 0;
}

input:placeholder-shown + label {
  cursor: pointer;
  width: fit-content;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  transform-origin: left bottom;
  transform: translate(0, 2.125rem) scale(1.5);
}

::-webkit-input-placeholder {
  opacity: 0;
  transition: inherit;
}

input:focus::-webkit-input-placeholder {
  opacity: 1;
}

input:not(:placeholder-shown) + label,
input:focus + label {
  transform: translate(0, 0) scale(1);
  cursor: pointer;
  color: #50dae2;
  font-size: 1.2rem;
  font-weight: 600;
}
.form-error {
  position: absolute;
  bottom: 0%;
  left: 0;
  font-size: 1.2rem;
  font-weight: 500;
  transform: translateY(100%);
  color: #d10157;
}
