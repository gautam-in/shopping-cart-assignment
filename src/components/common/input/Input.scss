$grey: #eaeaea;
$darkGrey: #acacac;
$lightBlue: #50dae4;
$pink: #d10157;

.floatLabel {
  display: flex;
  flex-direction: column;
  width: 280px;
  position: relative;
  &:focus-within {
    label {
      transform: translate(0, 12px) scale(0.8);
      color: $lightBlue;
    }
    input {
      border-color: $lightBlue;
    }
  }
  input {
    width: 100%;
    height: 40px;
    padding: 20px 0px 0 0;
    outline: 0;
    border-color: $darkGrey;
    border-style: solid;
    border-width: 0 0 1px 0;
    background: #fff;
  }
  label {
    &.active {
      transform: translate(0, 12px) scale(0.8);
      color: $lightBlue;
    }
    padding: 0 12px 0px 0px;
    color: $darkGrey;
    pointer-events: none;
    position: absolute;
    transform: translate(0, 26px) scale(1);
    transform-origin: top left;
    transition: all 0.2s ease-out;
  }
}
