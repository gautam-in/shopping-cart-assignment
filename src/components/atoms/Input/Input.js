import React from 'react';

import './Input.scss';

const Input = ({ errorLabel, ...props }) => {
  return (
    <section className='input-field'>
      <input {...props} />
      <label htmlFor={props.id}>{props.placeholder}</label>
      <span className='form-error'>{errorLabel}</span>
    </section>
  );
};

export default Input;
