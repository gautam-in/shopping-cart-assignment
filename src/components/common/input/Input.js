import React from "react";
import "./Input.scss";

const Input = (props) => {
  return (
    <div className="floatLabel">
      <input
        id={props.id}
        value={props.value}
        onChange={(e) => props.onChange(e)}
        type={props.type}
      />
      <label className={props.value !== "" ? "active" : ""} htmlFor={props.id}>
        {props.placeholder}
      </label>
    </div>
  );
};

Input.defaultProps = {
  type: "text",
};

export default Input;
