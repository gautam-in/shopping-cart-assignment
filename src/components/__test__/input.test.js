import React from 'react';
import ReactDom from 'react-dom';
import Input from "../input";
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import * as renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<Input></Input>, div)
})

it("renders textbox correctly", () => {
    const { getByTestId } = render(<Input></Input>)
    expect(getByTestId('input'));

})
it("matches snapshot", () => {
    const tree = renderer.create(<Input></Input>).toJSON();
    expect(tree).toMatchSnapshot();
})