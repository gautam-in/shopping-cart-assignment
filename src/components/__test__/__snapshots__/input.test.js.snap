// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="form-group"
  data-testid="input"
>
  <label
    className="bmd-label-floating"
  />
  <input
    aria-errormessage="d1"
    aria-invalid="true"
    aria-required="true"
    className="form-control"
    onChange={[Function]}
    required={true}
  />
  <span
    aria-live="assertive"
    id="d1"
    style={
      Object {
        "display": "none",
      }
    }
  />
</div>
`;
